### PR TITLE
issue-256

### PR DESCRIPTION
**Pull request by suyashdb**
added warning for participants with age 89 or 89+

### CODE REVIEW

File: utils/issues/list.js [hunk 1]
```diff
@@ -276,5 +276,10 @@ module.exports = {
         key: 'JSON_SCHEMA_VALIDATION_ERROR',
         severity: 'error',
         reason: 'JSON file is not formatted according the the schema.'
+    },
+    56: {
+        key: 'Participants age 89 or 89+',
```
Comment (chrisgorgo) [inline utils/issues/list.js]: "89 or higher"

File: utils/issues/list.js [hunk 2]
```diff
@@ -276,5 +276,10 @@ module.exports = {
...
+        severity: 'warning',
+        reason: 'Participants age is 89 or 89+. Tag them as 89+'
```
Comment (chrisgorgo) [inline utils/issues/list.js]: Give more explanation about why this should be changed (privacy concerns). Link to some more information for example: https://www.hhs.gov/hipaa/for-professionals/privacy/special-topics/de-identification/

File: validators/tsv.js [hunk 3]
```diff
@@ -142,6 +142,28 @@ module.exports = function TSV (file, contents, fileList, callback) {
...
     }
+  // check partcipants.tsv for age 89+
+    var headers = rows[0].split('\t');
+    var ageIndex = headers.indexOf('age');
+    if (file.name === 'participants.tsv'){
+      for (var i = 0; i < rows.length; i++) {
+        var row = rows[i];
+        var row = row.slice(0,-1)
+        var values = row.split('\t');
+        var ageIdColumn = headers.indexOf("age\r");
```
Comment (chrisgorgo) [inline validators/tsv.js]: Isn't this the same as `ageIndex` as defined in line 147?

File: validators/tsv.js [hunk 4]
```diff
@@ -142,6 +142,28 @@ module.exports = function TSV (file, contents, fileList, callback) {
...
+        age = values[ageIdColumn]
+        if (age == 89 || age > 89) {
```
Comment (chrisgorgo) [inline validators/tsv.js]: Isn't his the same as `age >= 89`?

File: validators/tsv.js [hunk 5]
```diff
@@ -142,6 +142,28 @@ module.exports = function TSV (file, contents, fileList, callback) {
...
+            age_89_plus = true;
```
Comment (chrisgorgo) [inline validators/tsv.js]: This variable is not used anywhere.

File: validators/tsv.js (hunk 4 above)
Comment (chrisgorgo) [reply validators/tsv.js]: Als don't you need to cast this to float to do this comparison?

File: validators/tsv.js [hunk 6]
```diff
@@ -142,6 +142,28 @@ module.exports = function TSV (file, contents, fileList, callback) {
         }
 
     }
+  // check partcipants.tsv for age 89+
+    var headers = rows[0].split('\t');
+    var ageIndex = headers.indexOf('age');
+    if (file.name === 'participants.tsv'){
+      for (var i = 0; i < rows.length; i++) {
+        var row = rows[i];
+        var row = row.slice(0,-1)
```
Comment (chrisgorgo) [inline validators/tsv.js]: Why is this necessary?

**Review by chrisgorgo — CHANGES_REQUESTED** [review]
Please add a test.

File: validators/tsv.js (hunk 6 above)
Comment (suyashdb) [reply validators/tsv.js]: as every row in tsv file ends with /r, sliced last two characters of every rows. 
For eg:  sub-01    group-A    20/r

**Comment by chrisgorgo** [comment]
Are you sure the end of the line characters are included? If so we need
take care of files created with UNIX systems (new line characters) and
windows (new line and carriage return characters). It might be better to
deal with this when you cast from string to a number.

**Comment by suyashdb** [comment]
It was the case for the test dataset(ds212) participant file, i have been
using. Let me check other tsv files too.

**Comment by suyashdb** [comment]
When i checked other tsv files, i observed that not all tsv files had EOL characters. Hence, first tried to deal with EOL characters while casting from string to Numbers. However, in case of header row, because in some files there was age\r or age\n and in others just age, i had to trim these EOL char while reading rows itself. 

For this reason, i went with the above logic (as per latest commit). 
Please let me know if you foresee any limitation in using this logic.

```
var header = rows[0].replace(/(\r\n|\n|\r)/gm,"").split('\t'); //remove EOL character if any
var ageIdColumn = header.indexOf("age");
```

I will add test in another commit.

**Comment by chrisgorgo** [comment]
This might be helpful https://www.w3schools.com/jsref/jsref_trim_string.asp

**Pull request merged**